src/main/java/bio/terra/service/filedata/azure/tables/TableDao.java

```diff
@@ -164,6 +165,27 @@ public FSItem retrieveByPath(
         fullPath);
   }
 
+  public Optional<FSItem> lookupOptionalPath(
+      UUID datasetId, String fullPath, AzureStorageAuthInfo storageAuthInfo, int enumerateDepth) {
+    TableServiceClient tableServiceClient =
+        azureAuthService.getTableServiceClient(
+            storageAuthInfo.getSubscriptionId(),
+            storageAuthInfo.getResourceGroupName(),
+            storageAuthInfo.getStorageAccountResourceName());
+    FireStoreDirectoryEntry fireStoreDirectoryEntry =
+        directoryDao.retrieveByPath(tableServiceClient, datasetId.toString(), fullPath);
```

Comment:
It looks like all uses of `directoryDao.retrieveByPath()` have a `UUID`, if so it would be better to accept `UUID` instead of `String` so callers don't need to convert it.
```suggestion
        directoryDao.retrieveByPath(tableServiceClient, datasetId, fullPath);
```